### PR TITLE
Houdini: instance in current context validator

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
Add optional validator check on invalid context data for asset and task in Houdini.

Many thanks to BigRoy https://github.com/ynput/ayon-core/issues/162#issuecomment-1985653191

## Testing notes:
1. Launch Houdini.
2. Create any instance and change its context to another asset/shot.
3. `ValidateInstanceInContextHoudini` should raise error and offer a repair action.
4. Click Repair Action, and it should work fine. 
4. This validator only error if there are any instances are pointing to different asset than the current.

### CODE REVIEW

**Comment by MustafaJafar** [comment]
Here you are a screenshot.
image

File: client/ayon_core/hosts/houdini/plugins/publish/validate_instance_in_context.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    label = "Instance in same Context"
+    optional = True
+    hosts = ["houdini"]
+    actions = [SelectROPAction, RepairAction]
+
+    def process(self, instance):
+        if not self.is_active(instance.data):
+            return
+
+        folderPath = instance.data.get("folderPath")
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/plugins/publish/validate_instance_in_context.py:34]: Variable `folderPath` should be `folder_path`

File: client/ayon_core/hosts/houdini/plugins/publish/validate_instance_in_context.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+                    "Usually this is not what you want but there can be cases "
+                    "where you might want to publish into another asset or "
+                    "shot. If that's the case you can disable the validation "
+                    "on the instance to ignore it."
+                )
+            )
+
+    @classmethod
+    def repair(cls, instance):
+        context_asset, context_task = cls.get_context(instance)
```
Comment (iLLiCiTiT) [inline client/ayon_core/hosts/houdini/plugins/publish/validate_instance_in_context.py:61]: Variable `context_asset` should be `context_folder`.

**Comment by moonyuet** [comment]
Tested in my side, when I switch to another folder path and task, it will raise the error as expected,
image

I will do the quick review one more time after the code tweaks.

**Review by BigRoy — APPROVED** [review]
Nice - looks good aside of Jakub's open notes.

**Review by moonyuet — APPROVED** [review]
image
Works smoothly

**Pull request merged**